### DISCUSSION TITLE
how to assign values to Numeric datatype

### DISCUSSION THREAD

**Discussion by vibhuh** [jackc/pgx#1880] [Q&A]
Hi,

I migrated to v 5.0 recently. Earlier I used like this:

numericDiscountPercentage := new(pgtype.Numeric)
numericDiscountPercentage.Set(updateCustomerObj.DiscountPercentage)

sqlc.UpdateCustomersParams{
SetDiscountPercentage:       !(updateCustomerObj.DiscountPercentage == 0),
DiscountPercentage:          *numericDiscountPercentage
}

Please help.

I Murugesan

**Comment by vibhuh** [comment]
The following works for me.

var numericDiscountPercentage pgtype.Numeric
	
if err := numericDiscountPercentage.Scan(strconv.FormatFloat(customerObj.DiscountPercentage, 'f', 2, 64)); err != nil {
	fmt.Println(err)
	return err
}

**Is this the best approach?**

**Reply by jackc** [reply]
That's as good a way as any -- if you need to use pgtype.Numeric. Personally, if I need to actually work with big decimal / numeric values I use the shopspring/decimal package. pgtype.Numeric is more of a necessary placeholder since the Go standard library doesn't have a big decimal type built in.
